routes/runtime/github.rb

```
@@ -48,7 +48,8 @@ class Clover
 
       fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil?
 
-      entry.update(last_accessed_at: Time.now, last_accessed_by: runner.id)
+      # Entry may no longer exist, use this.update to avoid raising an error in that case
+      entry.this.update(last_accessed_at: Sequel::CURRENT_TIMESTAMP, last_accessed_by: runner.id)
```

If it's already deleted, we return an invalid presigned URL since it silently ignores it. We should return a 204 in this case. 

If `.update` returns the number of updated rows, we can check that to make a decision.

---

`update` does return the number of updated rows.

I'm not sure if the `invalid presigned URL` is actually an issue. Without this change, the model instance could be deleted before the route returns, after the update.  What is the difference in the presigned URL between cases where the entry is deleted after the update as opposed to before?

If returning 204 fixes it, I'm fine with that approach, but then testing the conditional would be difficult.  We could just add `:nocov:`, but I'd prefer not to do that unless we are sure we should treat the cases differently.

---

Currently, when we encounter this case, we return an HTTP 500 error, and the client prints a warning about the unhandled exception.

With this PR, we no longer have an unhandled exception, but we are returning a presigned URL for a blob that has already been deleted. The client will attempt to download it, fail, and print again a warning about the failure.

The 204 status code is special for GitHub's cache client, and it doesn't print a warning for it. When we encounter this case, the cache entry no longer exists, so a 204 response is more appropriate for the situation.

What do you think about this approach?
```ruby
    if entry.this.update(last_accessed_at: Sequel::CURRENT_TIMESTAMP, last_accessed_by: runner.id) == 0
        fail CloverError.new(204, "NotFound", "No cache entry")
    end
```

OR

```ruby
    begin
        entry.update(last_accessed_at: Time.now, last_accessed_by: runner.id)
    rescue Sequel::NoExistingObject
        fail CloverError.new(204, "NotFound", "No cache entry")
    end
```

---

That approach makes sense to me.  I originally thought cache entry could be deleted after the UPDATE but before we return a result to the user, but since it updates last_accessed_at, I'm guessing that is not possible.

I'll work on this tomorrow.

---

Pushed a couple more commits.  One uses 204 in this case, with a slight restructuring to allow for full branch coverage.  While looking into this, I found what I think is a related race condition in the prog, so the other commit is a fix for that.